src/applications/mhv/secure-messaging/tests/e2e/sort-filter-test/secure-messaging-filter-inbox-folder.cypress.spec.js

```diff
@@ -0,0 +1,93 @@
+import PatientInboxPage from '../pages/PatientInboxPage';
+import SecureMessagingSite from '../sm_site/SecureMessagingSite';
+import mockMessages from '../fixtures/messages-response.json';
+
+describe('Secure Messaging Inbox Folder checks', () => {
+  const landingPage = new PatientInboxPage();
+  const site = new SecureMessagingSite();
+
+  const {
+    data: [, secondElement, thirdElement],
+    ...rest
+  } = mockMessages;
+
+  const mockFilterResults = { data: [secondElement, thirdElement], ...rest };
+
+  beforeEach(() => {
+    site.login();
+    landingPage.loadInboxMessages();
+  });
+
+  it('Verify filter works correctly', () => {
+    cy.injectAxe();
+    cy.axeCheck('main', {
+      rules: {
+        'aria-required-children': {
+          enabled: false,
+        },
+      },
+    });
+    landingPage.inputFilterData('test');
+    landingPage.filterMessages(mockFilterResults);
+    landingPage.verifyFilterResults('test', mockFilterResults);
+  });
+
+  it('Verify clear filter btn works correctly', () => {
+    cy.injectAxe();
+    cy.axeCheck('main', {
+      rules: {
+        'aria-required-children': {
+          enabled: false,
+        },
+      },
+    });
+    landingPage.inputFilterData('test');
+    landingPage.filterMessages(mockFilterResults);
+    landingPage.clearFilter();
+    landingPage.verifyFilterFieldCleared();
+  });
+
+  it('Check sorting works properly', () => {
+    cy.injectAxe();
+    cy.axeCheck('main', {
+      rules: {
+        'aria-required-children': {
+          enabled: false,
+        },
+      },
+    });
+    landingPage.verifySorting();
+  });
+});
+
+describe('Verify sorting feature with only one filter result', () => {
+  const landingPage = new PatientInboxPage();
+  const site = new SecureMessagingSite();
+
+  const {
+    data: [, secondElement],
+    ...rest
+  } = mockMessages;
+
+  const mockSingleFilterResult = { data: [secondElement], ...rest };
+
+  it('verify sorting not appears with only one filter result', () => {
```

Comment:
Just a small typo in the message here, could be "Verify sorting [does] not appear[...]".